search/search-api/api/src/java/org/sakaiproject/search/api/SearchService.java

```diff
@@ -156,6 +158,62 @@ SearchList search(String searchTerms, List<String> contexts, int searchStart,
 	 */
 	public SearchList search(String searchTerms, List<String> contexts, int start,
 			int end, String filterName, String sorterName) throws InvalidSearchQueryException;
+
+	/**
+	 * This is the same as standard search, but the caller can specify the logical name
+	 * of the index builder which should execute the search.
+	 *
+	 *
+	 * @param searchTerms A search string
+	 * @param contexts A list of contexts
+	 * @param start starting from
+	 * @param end ending at
+	 * @param indexBuilderName logical index builder name or {@code null} to fall back to default
+	 * @return
+	 * @throws InvalidSearchQueryException if unable to parse the query
+     */
+	SearchList search(String searchTerms, List<String> contexts, int start,
```

Comment:
Well. Multiple `SearchService` beans likely would have been a bigger impact to `search-tool` (though that's probably easy to overstate), and having a single `SearchService` bean maps cleanly to the underlying infrastructure (a single ES node regardless of how many index(builders)) and generally keeps interactions stable and simple for clients, especially if multi-index(builder) queries should be supported in the future (query fan-out and aggregation would go in the service rather than the client, just as is does for admin ops).

Agree, though, that the `search()` arg list is (and maybe already was) out of control. Maybe better to just add one `search(Map<String,Object> opts)` overload to accommodate this and any future query options.